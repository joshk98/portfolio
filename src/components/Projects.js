import React from "react";

import "../styles/projects.css";

const projectsData = [
  {
    id: 1,
    title: "Gourmet Guide",
    link: "https://github.com/joshk98/gourmet-guide",
    type: "App",
    src: "",
    alt: "",
    description: "",
  },
  {
    id: 2,
    title: "Breakout Game",
    link: "",
    type: "Game",
    src: "",
    alt: "",
    description: "",
  },
];

function Projects() {
  return (
    <div className="projects-container">
      {projectsData.map((project) => (
        <ul key={project.id} className="projects-container-card">
          <li className="projects-container-card-title">
            <a href={project.link} target="_blank" rel="noopener noreferrer">
              {project.title}
            </a>
          </li>
          <li className="projects-container-card-type">{project.type}</li>
          <li className="projects-container-card-image">
            <img src={project.src} alt={project.alt} />
          </li>
          <li className="projects-container-card-description">
            {project.description}
          </li>
        </ul>
      ))}
    </div>
  );
}

export default Projects;
