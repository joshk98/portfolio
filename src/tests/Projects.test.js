import React from "react";
import { render } from "@testing-library/react";
import Projects from "../components/Projects";

describe("Projects", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Projects />);

    expect(asFragment()).toMatchSnapshot();
  });
});
