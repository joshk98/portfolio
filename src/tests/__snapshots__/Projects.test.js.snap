// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects renders correctly 1`] = `
<DocumentFragment>
  <div
    class="projects-container"
  >
    <ul
      class="projects-container-card"
    >
      <li
        class="projects-container-card-title"
      >
        <a
          href="https://github.com/joshk98/gourmet-guide"
          rel="noopener noreferrer"
          target="_blank"
        >
          Gourmet Guide
        </a>
      </li>
      <li
        class="projects-container-card-type"
      >
        App
      </li>
      <li
        class="projects-container-card-image"
      >
        <img
          alt=""
          src=""
        />
      </li>
      <li
        class="projects-container-card-description"
      />
    </ul>
    <ul
      class="projects-container-card"
    >
      <li
        class="projects-container-card-title"
      >
        <a
          href=""
          rel="noopener noreferrer"
          target="_blank"
        >
          Breakout Game
        </a>
      </li>
      <li
        class="projects-container-card-type"
      >
        Game
      </li>
      <li
        class="projects-container-card-image"
      >
        <img
          alt=""
          src=""
        />
      </li>
      <li
        class="projects-container-card-description"
      />
    </ul>
  </div>
</DocumentFragment>
`;
